### PR TITLE
Feature/tracks front end

### PR DESCRIPTION
**Pull request by codyloyd**
this fleshes out the front-end for the tracks feature.

The work so far does not include any allowances for a slow-roll out... I just put all the features and links in place as if we were going to push it to everyone.

Things to check out/test:
- not logged in:
  - 'Curriculum' link in site header routes to our default track 'Full Stack Rails'
  - tracks#show includes a "Sign up" CTA and a link back to tracks#index
  - tracks#index lists the available tracks with their descriptions and a link to each #show page.
- logged in
  - dashboard displays only courses from selected track
  - dashboard includes link to tracks#index
  - tracks#show indicates either that this track is your currently selected one, or has a button that allows you to select it. (at the bottom of the page)

### CODE REVIEW

File: app/assets/stylesheets/tracks.scss [hunk 1]
```diff
@@ -1,3 +1,35 @@
 // Place all the styles related to the tracks controller here.
 // They will automatically be included in application.css.
 // You can use Sass (SCSS) here: http://sass-lang.com/
+
+.tracks-index {
```
Comment (KevinMulhern) [inline app/assets/stylesheets/tracks.scss]: Would it be worth bemify these styles to keep it consistent with our newer components?

Comment (codyloyd) [reply app/assets/stylesheets/tracks.scss]: probably

Comment (KevinMulhern) [reply app/assets/stylesheets/tracks.scss]: I won't let it block this, but I think we should make it a requirement going forward.

File: .env [hunk 2]
```diff
@@ -1,9 +1,9 @@
 API_USERNAME: 'development'
 API_PASSWORD: 'qwerty123'
-GITHUB_API_TOKEN: 1234
-GITHUB_APP_ID: 1234
-GITHUB_SECRET: 1234
-GOOGLE_CLIENT_ID: 1234
-GOOGLE_CLIENT_SECRET: 1234
-MAILCHIMP_API_KEY: 1234
-MAILCHIMP_LIST_ID: 1234
+GITHUB_API_TOKEN: da71593947eec1ad681da07a667cde6787003a90
```
Comment (mindovermiles262) [inline .env]: You might want to take this out and reset it

File: Gemfile [hunk 3]
```diff
@@ -29,6 +29,8 @@ gem 'nokogiri',                     '~> 1.8', '>= 1.8.4'
 gem 'sprockets',                    '~> 3.7.2'
 gem 'skylight'
 gem 'newrelic_rpm',               '~> 3.17'
+gem 'coffee-script'
```
Comment (arku) [inline Gemfile]: lol, why do we need this dependency?

File: Gemfile [hunk 4]
```diff
@@ -29,6 +29,8 @@ gem 'nokogiri',                     '~> 1.8', '>= 1.8.4'
...
+gem 'coffee-rails'
```
Comment (arku) [inline Gemfile]: This one too.

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -42,7 +42,7 @@
     <nav class="collapse navbar-collapse">
       <ul class="navbar-nav ml-auto">
         <li class="nav-item <%= active_class(courses_path) %>">
-          <%= link_to 'Curriculum', courses_path(ref: 'homenav'), class: 'nav-link' %>
+          <%= link_to 'Curriculum', track_path(1, ref: 'homenav'), class: 'nav-link' %>
```
Comment (arku) [inline app/views/shared/_navbar.html.erb]: What do you think of removing the hardcoded the track id and using `Track.default` instead?

```ruby
# app/models/track.rb

def self.default
  first
end
```

We can even go a step further and create a boolean column in the `tracks` table -  `default`. This gives us the flexibility to update the default course without changing the code.

WDYT?

File: app/views/tracks/show.html.erb [hunk 6]
```diff
@@ -1,8 +1,30 @@
...
+  <% else %>
+      <p class="text-center track-description">
+        <%= render 'shared/bottom_cta',
+          button: sign_in_or_view_curriculum_button,
+          heading: 'Start learning for free now!',
+          sub_heading: ''
+        %>
+      </p>
+  <% end %>
+  <p class="text-center track-description">You are viewing the <%= @track.title %> track.  To view all available tracks <%= link_to 'Click here', tracks_path %></p>
```
Comment (arku) [inline app/views/tracks/show.html.erb]:
```suggestion
  <p class="text-center track-description">You are viewing the <%= @track.title %> track.  To view all available tracks, <%= link_to 'click here', tracks_path %></p>
```

File: app/views/users/_skills.html.erb [hunk 7]
```diff
@@ -8,3 +8,5 @@
 <% courses.each do |course| %>
   <%= render 'skill', course: course %>
 <% end %>
+
+<p class="text-center">You are currently enrolled in the <%= @track.title %> track. To view all available tracks <%= link_to 'Click here', tracks_path %></p>
```
Comment (arku) [inline app/views/users/_skills.html.erb]:
```suggestion
<p class="text-center">You are currently enrolled in the <%= @track.title %> track. To view all available tracks, <%= link_to 'click here', tracks_path %></p>
```

File: config/routes.rb [hunk 8]
```diff
@@ -29,8 +29,11 @@
 
   # failure route if github information returns invalid
   get '/auth/failure' => 'omniauth_callbacks#failure'
-
   resources :users, only: [:show, :update]
+
+  namespace :users do
+    resources :tracks
```
Comment (arku) [inline config/routes.rb]: This creates all the CRUD routes.  Looks like we only need `POST create`.

Comment (arku) [inline config/routes.rb]:
```suggestion
    resources :tracks, only: %i(create)
```

File: .env (hunk 2 above)
Comment (arku) [inline .env]: 🙈 Yeah, please remove this token.

File: .env [hunk 9]
```diff
@@ -1,9 +1,9 @@
...
+# GITHUB_APP_ID: 1234
```
Comment (arku) [inline .env]: Why are all these commented out? 🤔

**Comment by arku** [comment]
@codyloyd Can you please update your branch to include changes from `master`?

**Pull request merged**